cli/src/semgrep/formatter/text.py

```diff
@@ -684,19 +684,31 @@ def format(
                 + "\n".join(first_party_nonblocking_output)
             )
         if first_party_blocking:
-            first_party_blocking_output = self._build_text_output(
-                first_party_blocking,
-                extra.get("color_output", False),
-                extra["per_finding_max_lines_limit"],
-                extra["per_line_max_chars_limit"],
-                extra["dataflow_traces"],
-            )
-            findings_output.append(
-                f"\nFirst-Party {blocking_description}:\n"
-                + "\n".join(first_party_blocking_output)
-            ) if (reachable or unreachable) else findings_output.append(
-                f"\n{blocking_description}:\n" + "\n".join(first_party_blocking_output)
-            )
+
+            def generate_output(header: str, matches: List[RuleMatch]) -> None:
+                if len(matches) > 0:
+                    output = self._build_text_output(
+                        matches,
+                        extra.get("color_output", False),
+                        extra["per_finding_max_lines_limit"],
+                        extra["per_line_max_chars_limit"],
+                        extra["dataflow_traces"],
+                    )
+                    findings_output.append(f"\n{header}:\n" + "\n".join(output))
+
+            if reachable or unreachable:
+                generate_output(
+                    f"First-Party {blocking_description}", first_party_blocking
+                )
+            else:
+                oss_matches = [
+                    x for x in first_party_blocking if not x.match.extra.is_pro_match
+                ]
+                pro_matches = [
+                    x for x in first_party_blocking if x.match.extra.is_pro_match
+                ]
+                generate_output(f"{blocking_description}", oss_matches)
+                generate_output(f"Semgrep PRO Findings", pro_matches)
```

Comment:
When we add `semgrep --deep` testing we should have a snapshot test for this output